test/index.js

```diff
@@ -222,6 +222,92 @@ describe('Lab', () => {
         expect(notebook.failures).to.equal(0);
     });
 
+    it('executes multiple pre/post processors and includes their context in tests', async () => {
+
+        const script = Lab.script({ schedule: false });
+        script.experiment('experiment', () => {
+
+            script.before(({ context }) => {
+
+                context.first = 1;
+            });
+
+            script.before(({ context }) => {
+
+                context.second = 2;
+            });
+
+            script.beforeEach(({ context }) => {
+
+                context.third = 3;
+            });
+
+            script.beforeEach(({ context }) => {
+
+                context.fourth = 4;
+            });
+
+            script.test('context is correct from befores', ({ context }) => {
+
+                expect(context.first).to.equal(1);
+                expect(context.second).to.equal(2);
+                expect(context.third).to.equal(3);
+                expect(context.fourth).to.equal(4);
+                context.test = true;
+            });
+
+            script.test('context cannot be manipulated', ({ context }) => {
+
+                expect(context.first).to.equal(1);
+                expect(context.second).to.equal(2);
+                expect(context.third).to.equal(3);
+                expect(context.fourth).to.equal(4);
+                expect(context.test).to.not.exist();
+            });
+
+            script.after(({ context }) => {
+
+                expect(context.third).to.equal(3);
+                expect(context.fourth).to.equal(4);
+                expect(context.test).to.not.exist();
+            });
+
+            script.afterEach(({ context }) => {
+
+                expect(context.first).to.equal(1);
+                expect(context.second).to.equal(2);
+                expect(context.third).to.equal(3);
+                expect(context.fourth).to.equal(4);
+                expect(context.test).to.not.exist();
```

Comment:
I'm not sure about this. If the assumption is that each test has its own context, I'd expect `context.test` to exist for one of the invocations here. I guess it could be hard to track individual tests across before, beforeEach, the test, after, and afterEach. Maybe this should be documented.

Reply:
I don't want a test to be able to manipulate the context, so keeping each separate for now... we can definitely change this if we want